vpn_slice/linux.py

```diff
@@ -108,3 +108,32 @@ def create_tunnel(self):
     def prepare_tunnel(self):
         if not os.access('/dev/net/tun', os.R_OK | os.W_OK):
             raise OSError("can't read and write /dev/net/tun")
+
+class LinuxSplitDNSProvider(SplitDNSProvider):
+    def __init__(self):
+        self.resolvconf = get_executable('/usr/bin/resolvconf')
+
+    def _resolvconf(self, *args):
+        cl = [self.resolvconf]
+        cl.extend(args)
+        subprocess.check_call(cl)
+
+    def configure_domain_vpn_dns(self, domains, nameservers, tundev):
+        cl = [self.resolvconf]
+        cl.extend(['-p', '-a', tundev])
+
+        p = subprocess.Popen(cl, stdin=subprocess.PIPE)
+
+        p.stdin.write("search ".encode('utf-8'))
+
+        for domain in domains:
+            p.stdin.write ((domain+" ").encode('utf-8'))
+        p.stdin.write("\n".encode('utf-8'))
+
+        for nameserver in nameservers:
+            p.stdin.write(("nameserver " + str(nameserver) + "\n").encode('utf-8'))
+
+        p.stdin.close()
```

Comment:
Please add a `p.wait()` so we don't move on until resolvconf is done.

Reply:
Ah, thanks for spotting this.